fs/walk.py

```diff
@@ -85,11 +87,22 @@ def __init__(
                 any of these patterns will be removed from the walk.
             filter_dirs (list, optional): A list of patterns that will be used
                 to match directories paths. The walk will only open directories
-                that match at least one of these patterns.
+                that match at least one of these patterns. Directories will
+                only be returned if the final component matches one of the
+                patterns.
             exclude_dirs (list, optional): A list of patterns that will be
                 used to filter out directories from the walk. e.g.
-                ``['*.svn', '*.git']``.
+                ``['*.svn', '*.git']``. Directories matching any of these
+                patterns will be removed from the walk.
             max_depth (int, optional): Maximum directory depth to walk.
+            filter_glob (list, optional): If supplied, this parameter
+                should be a list of path patterns e.g. ``["foo/**/*.py"]``.
+                Resources will only be returned if their global path or
+                an extension of it matches one of the patterns.
+            exclude_glob (list, optional): If supplied, this parameter
+                should be a list of path patterns e.g. ``["foo/**/*.py"]``.
```

Comment:
Not sure if it makes sense for the example patterns to be the same for both `filter_glob` and `exclude_glob`? :wink: Perhaps `["foo/**/*.pyc"]` would be a more useful example for the latter?